models/chainlink/arbitrum/chainlink_arbitrum_ccip_tokens_transferred.sql

```diff
@@ -0,0 +1,40 @@
+{{
+  config(
+    schema='chainlink_arbitrum',
+    alias='ccip_tokens_transferred',
+    materialized='incremental',
+    file_format='delta',
+    incremental_strategy='merge',
+    unique_key=['blockchain', 'tx_hash', 'total_tokens', 'token_symbol']
+  )
+}}
+
+
+SELECT DISTINCT
+    le.blockchain,
+    le.block_time,
+    le.tx_hash,
+    COALESCE(le.total_tokens / POWER(
+        CAST(10 AS double precision),
+        CAST(t.decimals AS double precision)
+    ), 0) AS total_tokens,
+    COALESCE(sr.destination_blockchain, 'UNKNOWN') AS destination_blockchain,
+    te.contract_address AS token,
+    COALESCE(t.symbol, 'UNKNOWN') AS token_symbol,
+    COALESCE(p.price * (
+        le.total_tokens / POWER(
+            CAST(10 AS double precision),
+            CAST(t.decimals AS double precision)
+        )
+    ), 0) AS token_price
+FROM
+    {{ref('chainlink_arbitrum_ccip_tokens_transferred_logs')}} le
+    INNER JOIN {{ref('chainlink_arbitrum_ccip_send_requested')}} sr ON le.tx_hash = sr.tx_hash
+    INNER JOIN  {{ source('erc20_arbitrum', 'evt_transfer') }} te ON le.contract_address = te.to
+    AND te.evt_tx_hash = le.tx_hash
+    LEFT JOIN {{ source('tokens', 'erc20') }} t ON te.contract_address = t.contract_address
+    LEFT JOIN {{ source('prices', 'usd') }} p ON te.contract_address = p.contract_address
+    AND p.minute = DATE_TRUNC ('minute', le.block_time)
+    {% if is_incremental() %}
+        WHERE {{ incremental_predicate('block_time') }}
+    {% endif %}
```

Comment:
any time you are filtering on a table incrementally, you should apply the same filter in join conditions to other tables being brought in. please apply incremental filter on all joins which contain data in time-series format (i.e. prices, evt_transfer, other chainlink refs, etc.)